File: ReactNativeApp/android/app/src/main/java/com/components/detection/person/PersonBuilder.java
package com.components.detection.person;

import com.models.ModelManager;
import com.ssvc.SSVC;
import com.ssvc.SSVCBuilder;

public class PersonBuilder extends SSVCBuilder {
    static PersonState buildState(ModelManager.MODELTYPE modelType) {
        return new PersonState(modelType);
    }

    public static SSVC<PersonState, PersonView, PersonController> buildPersonDetection(ModelManager.MODELTYPE modelType) {
        PersonState state = buildState(modelType);
        PersonView personView = new PersonView(state);
        PersonController personController = new PersonController(state);
        return new SSVC<>(state, personView, personController);
    }
}

File: ReactNativeApp/ios/Modules/SSVC/Instance/Detection/Person/PersonBuilder.swift
//
//  PersonBuilder.swift
//  jogo
//
//  Created by Mohsin on 18/10/2021.
//

import Foundation

internal class PersonBuilder: SSVCBuilder {
  
  internal static func buildState(modelType: ModelManager.MODELTYPE, exerciseLead: Bool) -> PersonState {
    return PersonState(modelType: ModelManager.MODELTYPE.POSENET, exerciseLead: false, observerId: Posenet.ObserverID.PERSON.rawValue)
  }
  
  internal static func build(state: PersonState) -> PersonSSVC<PersonState, PersonView, PersonController> {
    let view = PersonRectView(state: state)
    let controller = PersonController(state: state)
    return PersonSSVC(state: state, stateView: view, controller: controller)
  }
  
  internal static func build(modelType: ModelManager.MODELTYPE,
                                          exerciseLead: Bool) -> PersonSSVC<PersonState, PersonView, PersonController> {
    let state = buildState(modelType: modelType, exerciseLead: exerciseLead)
    let view = PersonRectView(state: state)
    let controller = PersonController(state: state)
    return PersonSSVC(state: state, stateView: view, controller: controller)
  }
}
